test/oauth2c_token_cache_gen_server_SUITE.erl

```diff
@@ -0,0 +1,45 @@
+-module(oauth2c_token_cache_gen_server_SUITE).
+-compile([export_all]).
+
+-include_lib("common_test/include/ct.hrl").
+-include_lib("stdlib/include/assert.hrl").
+
+
+all() -> [
+  get_token,
+  insert_token,
+  delete_token
+].
+
+init_per_suite(Config) -> Config.
+end_per_suite(_Config) -> ok.
+
+init_per_testcase(_TestCase, Config) ->
+  Now = os:system_time(millisecond),
+  TTL = 3.6e6,
+  InitialState = #{cache =>
+    #{key1 => {token1, Now}, key2 => {token2, Now - TTL}}, cache_ttl => TTL},
+  oauth2c_token_cache_gen_server:start(InitialState),
```

Comment:
Starting and stopping the process can be done in the `init/end_per_suite` functions instead.

Reply:
I understand your point, the reason why I choose to start and terminate the GenServer in the `init/end_pre_testcaste` was because I wanted to inject initial state and terminate the server at the end of each test case since I did not want to transfer the final state of a test case to the next (i.e. I wanted to start each test case with a clean state). Am I making a valid point or is this not important? :) Are there any better ways of achieving the same result?

Reply:
It's not a big issue, especially since this server is so simple and starts up in no time. You can leave it as it is.
When the startup is more complicated you would typically start in `init_per_suite` and stop in `end_per_suite`. It is then the responsibility of each `init_per_test_case` to set up the test case state and `end_per_test_case` to reset it afterwards. For instance, a positive test case for get would insert the element being checked in its `init_per_test_case`, and remove it in the `end_per_test_case`. The test cases should preferably be written so they do not depend on the state before `init_per_test_case`. This can be done by generating keys based on the test case name for instance.
One nice advantage of this is that test cases then can be run in parallel which reduces testing time.

Reply:
I guess I could test the GenServer by calling the callback methods directly instead of the user API (e.g. oauth2c_token_cache:handle_call({get, key}, State)) which makes it possible to inject the state directly when invoking the GenServer's callbacks and also prevents the need of even starting the gen_server at all when testing. However, I am not sure if this is an accepted way of testing GenServers?